torch/distributed/elastic/multiprocessing/api.py

```diff
@@ -470,11 +471,12 @@ def __init__(
 
     def start(self) -> None:
         """Start processes using parameters defined in the constructor."""
-        signal.signal(signal.SIGTERM, _terminate_process_handler)
-        signal.signal(signal.SIGINT, _terminate_process_handler)
-        if not IS_WINDOWS:
-            signal.signal(signal.SIGHUP, _terminate_process_handler)
-            signal.signal(signal.SIGQUIT, _terminate_process_handler)
+        if threading.current_thread() is threading.main_thread():
+            signal.signal(signal.SIGTERM, _terminate_process_handler)
+            signal.signal(signal.SIGINT, _terminate_process_handler)
+            if not IS_WINDOWS:
+                signal.signal(signal.SIGHUP, _terminate_process_handler)
+                signal.signal(signal.SIGQUIT, _terminate_process_handler)
```

Comment:
This has a lot of potential to cause issues since child processes won't be correctly cleaned up without this.

Is this being called programmatically? Can we pull this logic out into a helper function that can be explicitly called from the main thread so we keep this logic as well as support non-main threads?

Reply:
Actually reading this more -- we need torchelastic to run in the main thread as these handlers only run in the main thread